noisepagecontrol/exploratory_worker/services/data_collector/execute_data_collector.py

```diff
@@ -0,0 +1,47 @@
+import os
+import uuid
+
+from django.conf import settings
+
+from .data_collector_type import DataCollectorType
+from .catalog_data_collector.catalog_data_collector import CatalogDataCollector
+from .benchbase_data_collector.benchbase_data_collector import BenchbaseDataCollector
+
+from .create_data_archive import create_data_archive
+from .transfer_data import transfer_data
+
+DATA_COLLECTOR_MAP = {
+    DataCollectorType.CATALOG: CatalogDataCollector,
+    DataCollectorType.BENCHBASE: BenchbaseDataCollector,
+}
+
+
+def create_data_dir():
+    unique_identifier = str(uuid.uuid4())
+    data_dir_path = settings.DATA_COLLECTION_DIR / unique_identifier
+    os.mkdir(data_dir_path)
+    return data_dir_path
+
+
+def execute_data_collector(
+    event_name, resource_id, data_collector_type, postgres_port, config
+):
+
+    # create data dir
+    data_dir = create_data_dir()
+
+    # Init appopriate data collector
+    # TODO: Handle invalid data collector
+    data_collector = DATA_COLLECTOR_MAP[data_collector_type](
```

Comment:
nit: we can use importlib API + getattr so that people who want to add a DataCollector don't need to come here and modify DATA_COLLECTOR_MAP and data_collector_type.py

For example:
```
# assume data_collector_type = "CATALOG"

from importlib import import_module
data_collector_module = f"{data_collector_type.lower()}_data_collector"
module = import_module(data_collector_module)
data_collector = getattr(data_collector_module, toCamelCase(data_collector_module))

data_collector.setup()
data_collector.collect()
data_collector.cleanup()
...
```

Ref: https://stackoverflow.com/a/4605/12876831

Reply:
This is good but not on board of adding more complexity at this point; maybe in the future

Reply:
Also this would make validations difficult (whether a data collector exists or not)